### DISCUSSION TITLE
LightningCLI custom registry

### DISCUSSION THREAD

**Discussion by udeepam** [Lightning-AI/pytorch-lightning#12782] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Is there a way to add a custom registry to the LightningCLI?

Currently the LightningCLI instantiates classes from the following registries: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/cli.py#L107

What I would like to do is make a `CRITERION_REGISTRY` so that I can make custom criteria and make the criteria configurable from the config file. This would then be passed to the model.

**Comment by mauvilsa** [comment]
There is no need for registries to make a parameter (with type hint a class) configurable from the config file. In a config file you specify the full `class_path` and the respective `init_args` if any. The registries were only to make it convenient from the command line to specify a class giving only its name instead of the full class path. However, in a future version this will be possible to do without any need of explicitly registering the classes, see #12614. As consequence the registries will be deprecated soon, see #12961.
